src/linalg/blas.c

```diff
@@ -82,6 +82,85 @@ igraph_error_t igraph_blas_dgemv(igraph_bool_t transpose, igraph_real_t alpha,
     return IGRAPH_SUCCESS;
 }
 
+/**
+ * \function igraph_blas_dgemm
+ * \brief Matrix-matrix multiplication using BLAS.
+ *
+ * This function is a somewhat more user-friendly interface to
+ * the \c dgemm function in BLAS. \c dgemm performs the operation
+ * y = alpha*a*b + beta*c, where a, b and c are matrices, of which a and b
+ * can be transposed.
+ *
+ * \param transpose_a whether to transpose the matrix \p a
+ * \param transpose_b whether to transpose the matrix \p b
+ * \param alpha       the constant \c alpha
+ * \param a           the matrix \c a
+ * \param b           the matrix \c b
+ * \param beta        the constant \c beta
+ * \param c           the matrix \c c. The result will also be stored here.
+ *                    If beta is zero, c wil be resized to fit the result.
+ *
+ * Time complexity: O(nmk) where matrix a is of size n x k, and matrix b is of
+ * size k x m.
+ *
+ * \return \c IGRAPH_EOVERFLOW if the matrix is too large for BLAS,
+ *         \c IGRAPH_EINVAL if the matrices have incompatible sizes,
+ *         \c IGRAPH_SUCCESS otherwise.
+ *
+ * \example examples/simple/blas_dgemm.c
+ */
+igraph_error_t igraph_blas_dgemm(igraph_bool_t transpose_a, igraph_bool_t transpose_b,
+        igraph_real_t alpha, const igraph_matrix_t* a, const igraph_matrix_t* b,
+        igraph_real_t beta, igraph_matrix_t* c) {
+    char trans_a = transpose_a ? 'T' : 'N';
+    char trans_b = transpose_b ? 'T' : 'N';
+    int m, n, k, lda, ldb, ldc;
+    igraph_integer_t nrow_oa = transpose_a ? igraph_matrix_ncol(a) : igraph_matrix_nrow(a);
+    igraph_integer_t ncol_oa = transpose_a ? igraph_matrix_nrow(a) : igraph_matrix_ncol(a);
+    igraph_integer_t nrow_ob = transpose_b ? igraph_matrix_ncol(b) : igraph_matrix_nrow(b);
+    igraph_integer_t ncol_ob = transpose_b ? igraph_matrix_nrow(b) : igraph_matrix_ncol(b);
+
+    if (ncol_oa != nrow_ob) {
+        IGRAPH_ERROR("Number of columns of Matrix A not equal to "
+                "number of rows of matrix B (after possible transpositions of A and B).", IGRAPH_EINVAL);
+    }
+    if (beta && ncol_oa != igraph_matrix_ncol(c)) {
+        IGRAPH_ERROR("Number of columns of Matrix A not equal to "
+                "number of columns of matrix C (after possible transposition of A).", IGRAPH_EINVAL);
+    }
+    if (beta && nrow_oa != igraph_matrix_nrow(c)) {
+        IGRAPH_ERROR("Number of rows of Matrix A not equal to "
```

Comment:
Can you add a test case in the unit test that covers this branch? It seems like this one was missed.